_documentation/03-designing-scanner-drivers.md

```diff
@@ -2,5 +2,103 @@
 title: Designing Scanner Drivers
 toc: true
 toc_sticky: true
+h_range: [1,3]
 ---
-This document will contain a complete tutorial as well as information for manufacturers along examples for designing scanner drivers.
\ No newline at end of file
+**Currently PAPPL only provides framework for Printer Application. The codebase is currently under development and will be expanded in a future version of PAPPL to incorporate Multi Function Printer (MFP) functionalities such as scanning.** 
+
+**This document contains information for manufacturers for designing scanner drivers. Note that since the codebase is still under development, the document describes things as planned to be executed. In case of any changes, it will be updated as soon as possible. Also a complete tutorial along examples will be documented after the support for MFP functionalities is released.**
+
+## Introduction
+
+A driver is a code or data specific to a certain model or group of hardware devices, needed to make the hardware work with the hardware-model-independent code of the operating system. Printing in Linux has moved towards [Driverless Printing](https://openprinting.github.io/driverless/) and so does Scanning, which means there is no need for any hardware-model-specific code or data. 
+
+Kindly refer to <a href="../02-designing-printer-drivers/">Tutorial to Design Printer Drivers</a> know more about Printer Applications and their designing. Designing Scanner Applications is very much similar and expects the manufacturers to use PAPPL framework to reduce their effort in implementing all things from scratch.
+
+## Working of PAPPL-based Scanner Driver
+
+PAPPL provides support for the below mentioned commands by default:
+
+* add
+* cancel
+* default
+* delete
+* devices
+* drivers
+* jobs
+* modify
+* options
+* printers
+* scan
+* server
+* shutdown
+* status
+* submit
+
+Once the User issues any of the following Sub-command, The printer/scanner application converts these commands (or corresponding IPP requests) into the necessary USB/network/etc. communications to perform a scan, get capabilities, and so forth.
+
+The class of device that we are supporting for PAPPL, the scanner will only produce raster data. It might provide native `TIFF` support, but that is probably about it.
```

Comment:
**In the** class of device that we are supporting **with** PAPPL, the scanner will only produce raster data. It might provide native `TIFF` support, but that is probably **all** about it. **PDF for example is not supported**.